layers/+lang/emacs-lisp/packages.el

```diff
@@ -80,7 +80,7 @@
           (lisp-indent-line))))))
 
 (defun emacs-lisp/post-init-company ()
-  (spacemacs|add-company-backends :backends company-capf
+  (spacemacs|add-company-backends :backends (company-capf company-dabbrev-code)
```

Comment:
I am not sure that we really need this, I have copied your sample code into a sample buffer and tried to auto-complete test-fun, which did work very well after me having evaluated the new function. This behaviour did not change for me when I evaluate the new code.

Can you double check?

Reply:
Yes, I know that, after evaluating, the auto-completion works. But evaluating should not be necessary. When I type a single function and use the same new symbol in it a few times, it should get auto-completed, and currently it does not.
This is exactly what the `company-dabbrev` backend is for, and it is already configured for `emacs-lisp-mode` (actually it is in the Spacemacs default company backends). However, the configuration is wrong for `emacs-lisp-mode` because in this mode all completion is hijacked by `company-capf`. In other modes, where `company-capf` uses different auto-completion functions (e.g. pyhton mode), `company-capf` does not hijack all auto-completion, so for such modes `company-capf` is configured correctly. I actually, "discussed" it in a company issue, and this solution was what they recommended (and indeed it works fine and does what it should do here).

Reply:
Okay, I wanted to create an example, and it turns out that `comapny-capf` does complete `variables within scope`.
So I have quickly made up a little more exotic, but still very relevant, example:
```
;; This is a comment about this-example-file
(defun test-function (some-list)
  (if (eq some-list nil)
      nil
    (test-f

;; this-example-f
```
So here, even though I am in the scope of test-function, it does not yet get auto-completed. Of course here is no way that I can already evaluate the function before recursively using it.
Additionally, I have added a comment mentioning already e.g. the file-name, then when I try to make a second comment the file-name does not get completed. For sure there are more examples, I started this fix because it was annoying me, and then when I tested e.g. in vim (but probably also most other editors), it just worked as expected.